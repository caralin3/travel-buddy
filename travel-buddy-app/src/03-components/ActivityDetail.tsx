import moment from 'moment';
import React from 'react';
import { Col, Row } from 'reactstrap';
import { Activity } from '../api';
import { getDurationHours } from '../utils';

export interface ActivityDetailProps {
  activities: Activity[];
}

export const ActivityDetail: React.FC<ActivityDetailProps> = ({ activities }) => {
  const getDuration = (activity: Activity) => getDurationHours(activity.startDate, activity.endDate);

  return (
    <div className="activity-detail">
      <Row>
        <Col className="activity-detail__name text-primary">Activities</Col>
      </Row>
      <Row className="my-2">
        <Col xs={12} sm={2}>
          <p className="activity-detail__label">Time</p>
        </Col>
        <Col xs={12} sm={4}>
          <p className="activity-detail__label">Name</p>
        </Col>
        <Col xs={12} sm={6}>
          <p className="activity-detail__label">Notes</p>
        </Col>
      </Row>
      {activities.map((activity) => (
        <Row key={activity.id} className="activity">
          <Col xs={12} sm={2}>
            {moment(activity.startDate).format('h:mm A')}{' '}
            <em className="activity__duration">
              ({getDuration(activity)} hr{getDuration(activity) > 1 ? 's' : ''})
            </em>
          </Col>
          <Col xs={12} sm={4}>
            <p className="activity__name text-primary m-0">
              {activity.name}
              {!!activity.port && (
                <span className="activity__port">
                  &nbsp;({activity.port.city}, {!!activity.port.state ? activity.port.state : activity.port.country})
                </span>
              )}
            </p>
            <p className="activity__address m-0">
              {activity.addressLine1}
              {!!activity.addressLine2 ? ` ${activity.addressLine2}` : ''}
              {!!activity.city ? `, ${activity.city}` : ''}
              {!!activity.state ? `, ${activity.state}` : ''}
              {/* {!!activity.postalCode ? activity.postalCode : ''}{' '} */}
            </p>
          </Col>
          <Col xs={12} sm={6}>
            <p className="activity__notes m-0">{activity.description}</p>
          </Col>
        </Row>
      ))}
    </div>
  );
};
