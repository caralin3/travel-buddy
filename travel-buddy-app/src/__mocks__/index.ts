export * from './activities';
export * from './cruises';
export * from './flights';
export * from './hotels';
export * from './trips';
export * from './user';
