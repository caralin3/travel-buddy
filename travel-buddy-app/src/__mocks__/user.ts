import { User } from '../api';

export const user: User = {
  id: 0,
  firstName: 'Jane',
  lastName: 'Doe',
  email: 'janeD@email.com',
  roles: [
    {
      id: 0,
      name: 'ROLE_USER',
    },
  ],
};
