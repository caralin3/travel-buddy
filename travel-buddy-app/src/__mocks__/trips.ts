import { Trip } from '../api';
import { user } from './user';

export const trips: Trip[] = [
  {
    user,
    id: 0,
    title: 'Texas Girls Trip 2022',
    description: 'Travelers: Cara, Elizabeth, Sarah<br/>Destination: Dallas, Texas',
    startDate: '2022-06-03',
    endDate: '2022-06-06',
    uniqueLink: '',
  },
  {
    user,
    id: 1,
    title: 'Trip 2',
    startDate: '2022-08-14',
    endDate: '2022-08-31',
  },
];
