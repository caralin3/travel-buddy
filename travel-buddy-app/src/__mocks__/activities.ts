import { Activity } from '../api';
import { trips } from './trips';

export const activities: Activity[] = [
  {
    id: 0,
    name: 'Horseback Riding',
    description: 'Horseback riding at Stockyard stables.',
    startDate: '2022-06-04 11:00:00',
    endDate: '2022-06-04 12:00:00',
    cost: 75,
    currency: 'USD',
    addressLine1: '157 E Exchange Ave',
    city: 'Fort Worth',
    state: 'Texas',
    country: 'USA',
    postalCode: '76164',
    trip: trips[0],
  },
  {
    id: 1,
    name: 'Billy Bobs Texas',
    description: 'Honky Tonk near Stockyard stables.',
    startDate: '2022-06-04 16:00:00',
    endDate: '2022-06-04 18:00:00',
    cost: 20,
    currency: 'USD',
    addressLine1: '2520 Rodeo Plaza',
    city: 'Fort Worth',
    state: 'Texas',
    country: 'USA',
    postalCode: '76164',
    trip: trips[0],
  },
  {
    id: 2,
    name: 'The Sixth Floor Museum at Dealey Plaza',
    description: 'JFK Museum',
    startDate: '2022-06-05 13:30:00',
    endDate: '2022-06-05 15:00:00',
    cost: 1,
    currency: 'USD',
    addressLine1: '411 Elm Street',
    city: 'Dallas',
    state: 'Texas',
    country: 'USA',
    postalCode: '75202',
    trip: trips[0],
  },
  {
    id: 3,
    name: "Adair's Saloon",
    description:
      'Same area=Dallas World Aquarium, 6th floor museum and dealy plaza, and Museum of Nature and Science SE of hotel ',
    startDate: '2022-06-03 16:00:00',
    endDate: '2022-06-03 18:00:00',
    cost: 20,
    currency: 'USD',
    addressLine1: '2624 Commerce Street',
    city: 'Dallas',
    state: 'Texas',
    country: 'USA',
    postalCode: '75226',
    trip: trips[0],
  },
];
