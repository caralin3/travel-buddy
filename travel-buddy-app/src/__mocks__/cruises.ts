import { Cruise, Port } from '../api';
import { trips } from './trips';

export const cruises: Cruise[] = [
  {
    id: 0,
    cabinType: 'SUITE',
    cruiseLine: 'Princess Cruises',
    shipName: 'Discovery Princess',
    cost: 3000,
    currency: 'USD',
    cabinNumber: '11234',
    startDate: '2022-05-08 16:00:00',
    endDate: '2022-05-15 07:00:00',
    description: 'Alaskan Cruise',
    departureCity: 'Seattle',
    departureState: 'Washington',
    departureCountry: 'US',
    roundTrip: true,
    trip: trips[0],
  },
  {
    id: 1,
    cabinType: 'DOUBLE',
    cruiseLine: 'Royal Caribbean',
    shipName: 'Discovery Princess',
    cost: 5402.97,
    currency: 'USD',
    cabinNumber: '9112',
    startDate: '2025-04-12',
    endDate: '2025-04-19',
    departureCity: 'Sicily',
    departureCountry: 'Italy',
    destinationCity: 'Athens',
    destinationCountry: 'Greece',
    roundTrip: false,
    trip: trips[0],
  },
];

export const cruise1Ports: Port[] = [
  {
    id: 0,
    day: 2,
    arrival: '2022-05-10 06:30:00',
    departure: '2022-05-10 15:00:00',
    city: 'Ketchikan',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 1,
    day: 3,
    arrival: '2022-05-11 12:30:00',
    departure: '2022-05-11 21:00:00',
    city: 'Juneau',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 2,
    day: 4,
    arrival: '2022-05-12 06:00:00',
    departure: '2022-05-12 17:00:00',
    city: 'Skagway',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 3,
    day: 6,
    arrival: '2022-05-14 19:00:00',
    departure: '2022-05-14 23:59:00',
    city: 'Victoria',
    country: 'Canada',
    cruise: cruises[0],
  },
];

export const cruise2Ports: Port[] = [
  {
    id: 0,
    day: 2,
    arrival: '2025-04-14 06:30:00',
    departure: '2025-04-14 15:00:00',
    city: 'Florence',
    country: 'Italy',
    cruise: cruises[1],
  },
  {
    id: 1,
    day: 5,
    arrival: '2025-04-17 12:30:00',
    departure: '2025-04-17 21:00:00',
    city: 'Rome',
    country: 'Italy',
    cruise: cruises[1],
  },
  {
    id: 2,
    day: 6,
    arrival: '2025-04-18 06:00:00',
    departure: '2025-04-18 17:00:00',
    city: 'Santorini',
    country: 'Greece',
    cruise: cruises[1],
  },
];
