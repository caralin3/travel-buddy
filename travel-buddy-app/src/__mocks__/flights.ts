import { Flight } from '../api';
import { trips } from './trips';

export const flights: Flight[] = [
  {
    id: 1,
    airline: 'JetBlue',
    arrivalAirport: 'JFK',
    arrivalCity: 'NYC',
    arrivalDate: '2022-06-03 15:30:00',
    confirmationCode: 'FWD5DBJ',
    departureAirport: 'DFW',
    departureCity: 'Dallas',
    departureDate: '2022-06-06 13:05:00',
    flightClass: 'ECONOMY',
    flightNumber: '124',
    seats: '12 DEF',
    terminal: 'E',
    trip: trips[0],
  },
  {
    id: 0,
    airline: 'JetBlue',
    arrivalAirport: 'DFW',
    arrivalCity: 'Dallas',
    arrivalDate: '2022-06-03 13:39:00',
    confirmationCode: 'FWD5DBJ',
    cost: 300.03,
    currency: 'USD',
    departureAirport: 'JFK',
    departureCity: 'NYC',
    departureDate: '2022-06-03 10:05:00',
    flightClass: 'ECONOMY',
    flightNumber: '123',
    terminal: '5',
    seats: '7 ABC',
    trip: trips[0],
  },
];
