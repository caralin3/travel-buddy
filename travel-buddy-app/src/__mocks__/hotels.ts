import { Hotel } from '../api';
import { trips } from './trips';

export const hotels: Hotel[] = [
  {
    id: 0,
    name: 'Courtyard By Marriot Denton',
    roomType: 'DOUBLE',
    roomCount: 1,
    checkInDate: '2022-06-03 15:00:00',
    checkOutDate: '2022-06-06 11:00:00',
    cost: 0,
    currency: 'USD',
    addressLine1: '2800 Colorado Blvd',
    city: 'Denton',
    state: 'TX',
    trip: trips[0],
  },
];
