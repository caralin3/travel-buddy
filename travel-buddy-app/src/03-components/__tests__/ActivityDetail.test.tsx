import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { activities } from '../../__mocks__';
import { ActivityDetail } from '../ActivityDetail';

describe('ActivityDetail', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ActivityDetail activities={activities} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
