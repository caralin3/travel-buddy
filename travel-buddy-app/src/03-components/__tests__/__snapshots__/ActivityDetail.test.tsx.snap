// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityDetail should render correctly 1`] = `
<div
  className="activity-detail"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      className="activity-detail__name text-primary"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      Activities
    </Col>
  </Row>
  <Row
    className="my-2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity-detail__label"
      >
        Time
      </p>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity-detail__label"
      >
        Name
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity-detail__label"
      >
        Notes
      </p>
    </Col>
  </Row>
  <Row
    className="activity"
    key="0"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      11:00 AM
       
      <em
        className="activity__duration"
      >
        (
        1
         hr
        )
      </em>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__name text-primary m-0"
      >
        Horseback Riding
      </p>
      <p
        className="activity__address m-0"
      >
        157 E Exchange Ave
        , Fort Worth
        , Texas
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__notes m-0"
      >
        Horseback riding at Stockyard stables.
      </p>
    </Col>
  </Row>
  <Row
    className="activity"
    key="1"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      4:00 PM
       
      <em
        className="activity__duration"
      >
        (
        2
         hr
        s
        )
      </em>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__name text-primary m-0"
      >
        Billy Bobs Texas
      </p>
      <p
        className="activity__address m-0"
      >
        2520 Rodeo Plaza
        , Fort Worth
        , Texas
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__notes m-0"
      >
        Honky Tonk near Stockyard stables.
      </p>
    </Col>
  </Row>
  <Row
    className="activity"
    key="2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      1:30 PM
       
      <em
        className="activity__duration"
      >
        (
        1.5
         hr
        s
        )
      </em>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__name text-primary m-0"
      >
        The Sixth Floor Museum at Dealey Plaza
      </p>
      <p
        className="activity__address m-0"
      >
        411 Elm Street
        , Dallas
        , Texas
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__notes m-0"
      >
        JFK Museum
      </p>
    </Col>
  </Row>
  <Row
    className="activity"
    key="3"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      4:00 PM
       
      <em
        className="activity__duration"
      >
        (
        2
         hr
        s
        )
      </em>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__name text-primary m-0"
      >
        Adair's Saloon
      </p>
      <p
        className="activity__address m-0"
      >
        2624 Commerce Street
        , Dallas
        , Texas
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__notes m-0"
      >
        Same area=Dallas World Aquarium, 6th floor museum and dealy plaza, and Museum of Nature and Science SE of hotel 
      </p>
    </Col>
  </Row>
  <Row
    className="activity"
    key="4"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      4:00 PM
       
      <em
        className="activity__duration"
      >
        (
        2
         hr
        s
        )
      </em>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__name text-primary m-0"
      >
        Arrive in Seattle
      </p>
      <p
        className="activity__address m-0"
      >
        Seattle Washington
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__notes m-0"
      >
        Arrive for cruise
      </p>
    </Col>
  </Row>
  <Row
    className="activity"
    key="5"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      4:00 PM
       
      <em
        className="activity__duration"
      >
        (
        2
         hr
        s
        )
      </em>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__name text-primary m-0"
      >
        After Dinner Show
      </p>
      <p
        className="activity__address m-0"
      >
        Main Theater
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__notes m-0"
      >
        Show after dinner
      </p>
    </Col>
  </Row>
  <Row
    className="activity"
    key="6"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      sm={2}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      8:00 AM
       
      <em
        className="activity__duration"
      >
        (
        2
         hr
        s
        )
      </em>
    </Col>
    <Col
      sm={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__name text-primary m-0"
      >
        Helicopter Ride
        <span
          className="activity__port"
        >
           (
          Ketchikan
          , 
          Alaska
          )
        </span>
      </p>
      <p
        className="activity__address m-0"
      >
        2624 Commerce Street
        , Ketchikan
        , Alaska
      </p>
    </Col>
    <Col
      sm={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={12}
    >
      <p
        className="activity__notes m-0"
      >
        Helicopter ride over glaciers
      </p>
    </Col>
  </Row>
</div>
`;
